### ISSUE TITLE
[BUG] Duplicated tracks on favorites view

### ISSUE THREAD

**Issue by Pitros** [Soundnode/soundnode-app#564]
Am I only having those duplicated tracks after scrolling down and new likes being added?

![image](https://cloud.githubusercontent.com/assets/6910670/12658678/c5bf617a-c60a-11e5-9c79-02093542dae7.png)

**Comment by wb9688** [comment]
Nope, I also have it sometimes...
Op 28 jan. 2016 22:03 schreef "Piotr Cholewa" <notifications@github.com>:

**Comment by yeahoffline** [comment]
same here

**Comment by ELKOKACOLO** [comment]
Same here

**Comment by velagand** [comment]
Yup, I have this as well. Any thoughts on why this is happening?

**Comment by zestyr** [comment]
This bug is still prevalent under likes and when viewing an artist's songs. Stream and Top 50 works fine though.

**Comment by weblancaster** [comment]
I need to investigate that @zestyr

**Comment by zestyr** [comment]
Not sure if it matters, but I experience the issue on both Arch Linux and Linux Mint.

**Comment by weblancaster** [comment]
Thanks for reporting but yeah it's going to happen in every platform @zestyr

**Comment by zestyr** [comment]
Hmm, could [this](https://github.com/Soundnode/soundnode-app/blob/9e60387b708c95405f32bb19142f551faea475fa/app/public/js/favorites/favoritesCtrl.js#L35) have anything to do with it? When the next page is loaded, it gets added twice (albeit into two different lists).

``` javascript       
SCapiService.getNextPage()
    .then(function(data) {
        for ( var i = 0; i < data.collection.length; i++ ) {
            $scope.originalData.push( data.collection[i] );
            $scope.data.push( data.collection[i] )
        }
        utilsService.updateTracksReposts(data.collection, true);
    },
```

**Comment by Pitros** [comment]
@zestyr yep, that's the problem. I made a PR with fix #591 a while ago, but...
